community/planning/challenge_authorization.md

```diff
@@ -296,33 +288,74 @@ message AuthorizationError {
 ## Rationale and alternatives
 [rationale-and-alternatives]: #rationale-and-alternatives
 
-This design includes adding a list of public keys to the node definition that
-will be used for Challenge Authorization. One alternative would be to link a
-public key to a specific endpoint in the node definition.
+This design includes adding a public key to the member definition that will be
+used for Challenge Authorization. One alternative would be to link a public key
+to a specific endpoint in the node definition.
 
 ```yaml
 nodes:
     acme-node-000:
         id: acme-node-000
         endpoints:
-            endpoint-1:
-              challenge_public_key: PUBLIC_KEY
-              endpoint: "tcps://splinterd-node-acme:8044"
-            endpoint-2:
-              challenge_public_key: PUBLIC_KEY
-              endpoint: "tcps://splinterd-node-acme-2:8074"
+      endpoint-1:
+        challenge_public_key: PUBLIC_KEY
+        endpoint: "tcps://splinterd-node-acme:8044"
+      endpoint-2:
+        challenge_public_key: PUBLIC_KEY
+        endpoint: "tcps://splinterd-node-acme-2:8044"
+
+```
+
+Instead of setting one authorization, it could be updated to be a list of
+authorization types in preference order. This would allow a circuit to be
+updated to support both authorization types at the same time. This, however,
+would require two separate updates to the circuit, the first to add the new
+authorization type, and another to remove the old authorization type after all
+nodes have updated to support the new authorization type. Instead, the
+authorization update should only be allowed if the node can support the new
+authorization type.
+
+```yaml
+circuits:
+    WBKLF-AAAAA:
+        id: WBKLF-AAAAA
+    +   authorization: ["Challenge", "Trust"]
+```
+
+Similarly, a member in a circuit could have multiple public keys but instead the
+splinter daemon will support multiple signing keys at once but only include one
+key in a circuit. This will allow a node to update its key in a circuit, while
+still supporting its old key in other circuits.
 
+Instead of submitting multiple submit requests for all of the supported keys,
+only one public key and signature could be returned. This however would require
+a more the expected public key to be returned with the nonce and this
```

Comment:
Seems like slight miswording here, not sure what it _should_ be

Reply:
fixed